ast/case_statement.go

```diff
@@ -0,0 +1,53 @@
+package ast
+
+import (
+	"bytes"
+)
+
+type CaseStatement struct {
+	*Meta
+	Test       *InfixExpression
+	Statements []Statement
+}
+
+func (s *CaseStatement) statement()     {}
+func (s *CaseStatement) GetMeta() *Meta { return s.Meta }
+func (s *CaseStatement) FinalStatement() Statement {
+	return s.Statements[len(s.Statements)-1]
+}
+func (s *CaseStatement) IsBreak() bool {
+	_, ok := s.FinalStatement().(*BreakStatement)
+	return ok
+}
+func (s *CaseStatement) IsFallthrough() bool {
+	_, ok := s.FinalStatement().(*FallthroughStatement)
+	return ok
+}
```

Comment:
`AST` must have only syntax tree and token representation so we should not have any logic here.
An `interpreter` package should implement this logic